cf/test/test_subsampling.py

```diff
@@ -0,0 +1,249 @@
+        self.assertTrue(
+            lon.equals(lon0, ignore_data_type=True, rtol=0, atol=1e-25)
+        )
+
+        # Bounds
+        lat0 = f[6].data
+        lon0 = f[10].data
+
+        lat = q.construct("latitude").bounds.data
+        lon = q.construct("longitude").bounds.data
+
+        self.assertEqual(lat.dtype, float)
+        self.assertEqual(lon.dtype, float)
+
+        self.assertTrue(
+            lat.equals(lat0, ignore_data_type=True, rtol=0, atol=1e-13)
+        )
+        self.assertTrue(
+            lon.equals(lon0, ignore_data_type=True, rtol=0, atol=1e-13)
+        )
+
+    def test_bi_linear(self):
+        """Test bi-linear interpolation."""
+        f = cf.read(self.bilinear)
+
+        q = f[0]
+        lat0 = f[1].data
+        lon0 = f[3].data
+
+        lat = q.construct("ncvar%a_2d").data
+        lon = q.construct("ncvar%b_2d").data
+
+        self.assertEqual(lat.dtype, float)
+        self.assertEqual(lon.dtype, float)
+
+        self.assertTrue(
+            lat.equals(lat0, ignore_data_type=True, rtol=0, atol=1e-25)
+        )
+        self.assertTrue(
+            lon.equals(lon0, ignore_data_type=True, rtol=0, atol=1e-25)
+        )
+
+        # Bounds
+        lat0 = f[2].data
+        lon0 = f[4].data
+
+        lat = q.construct("ncvar%a_2d").bounds.data
+        lon = q.construct("ncvar%b_2d").bounds.data
+
+        self.assertEqual(lat.dtype, float)
+        self.assertEqual(lon.dtype, float)
+
+        self.assertTrue(
+            lat.equals(
+                lat0, ignore_data_type=True, rtol=0, atol=1e-13, verbose=-1
+            )
+        )
+        self.assertTrue(
+            lon.equals(
+                lon0, ignore_data_type=True, rtol=0, atol=1e-13, verbose=-1
+            )
+        )
+
+    def test_quadratic(self):
+        """Test quadratic interpolation."""
+        f = cf.read(self.quadratic)
+
+        t = f[-2]
+        lat0 = f[7].data
+        lon0 = f[11].data
+
+        lat = t.construct("latitude").data
+        lon = t.construct("longitude").data
+
+        self.assertEqual(lat.dtype, float)
+        self.assertEqual(lon.dtype, float)
+
+        self.assertTrue(
+            lat.equals(lat0, ignore_data_type=True, rtol=0, atol=1e-25)
+        )
+        self.assertTrue(
+            lon.equals(lon0, ignore_data_type=True, rtol=0, atol=2e-5)
+        )
+
+        # Bounds
+        lat0 = f[8].data
+        lon0 = f[12].data
+
+        lat = t.construct("latitude").bounds.data
+        lon = t.construct("longitude").bounds.data
+
+        self.assertEqual(lat.dtype, float)
+        self.assertEqual(lon.dtype, float)
+
+        self.assertTrue(
+            lat.equals(
+                lat0, ignore_data_type=True, rtol=0, atol=9e-5, verbose=-1
+            )
+        )
+        self.assertTrue(
+            lon.equals(
+                lon0, ignore_data_type=True, rtol=0, atol=9e-5, verbose=-1
+            )
+        )
+
+    @unittest.skipIf(True, "TODO: awaiting test file")
```

Comment:
Any news on when this test file might become available? I seem to recall (for the `cfdm` case, I guess the same file will be used for both libraries) you asked someone to provide something for this specific case and were waiting on that (though that could be my memory going wrong)?

Reply:
Ah yes - still waiting, although I have been in touch. I'll chase it up again ...

Reply:
Thanks for clarifying. Obviously it would be good to not leave it too long else it might fall off our radar and then never get tested overall...